synthesizer/program/src/logic/instruction/operation/hash.rs

```diff
@@ -196,6 +203,120 @@ impl<N: Network, const VARIANT: u8> HashInstruction<N, VARIANT> {
+    output: Literal<N>,
+    destination_type: &PlaintextType<N>,
+) -> Result<Value<N>> {
+    let dest = match destination_type {
+        PlaintextType::Literal(literal_type) => output.cast_lossy(*literal_type)?,
+        PlaintextType::Struct(..) => bail!("Cannot hash into a struct"),
+        PlaintextType::Array(..) => bail!("Cannot hash into an array (yet)"),
+    };
+
+    Ok(Value::Plaintext(Plaintext::from(dest)))
+}
+
+fn to_destination_circuit_literal<A: circuit::Aleo, N: Network>(
+    output: circuit::Literal<A>,
+    destination_type: &PlaintextType<N>,
+) -> Result<circuit::Value<A>> {
+    let dest = match destination_type {
+        PlaintextType::Literal(literal) => output.cast_lossy(*literal)?,
+        PlaintextType::Struct(..) => bail!("Cannot hash into a struct"),
+        PlaintextType::Array(..) => bail!("Cannot hash into an array (yet)"),
+    };
+
+    Ok(circuit::Value::Plaintext(circuit::Plaintext::Literal(dest, Default::default())))
+}
+
+fn to_console_array<N: Network>(hash: &[bool], len: usize, literal_type: LiteralType) -> Result<Plaintext<N>> {
+    let chunk_size = match literal_type {
+        LiteralType::U8 => 8,
+        LiteralType::U16 => 16,
+        _ => bail!("Unsupported literal type: {literal_type} for array"),
+    };
+
+    ensure!(
+        hash.len() % chunk_size == 0,
+        "Expected hash length to be a multiple of {chunk_size}, but found {}",
+        hash.len()
+    );
+
+    let hash = hash
+        .chunks(chunk_size)
+        .map(|element| {
+            let value = element.iter().enumerate().fold(0, |acc, (i, bit)| if *bit { acc | (1 << i) } else { acc });
+
+            match literal_type {
+                LiteralType::U8 => {
+                    Ok(Plaintext::Literal(Literal::U8(console::types::U8::new(value.try_into()?)), Default::default()))
+                }
+                LiteralType::U16 => Ok(Plaintext::Literal(
+                    Literal::U16(console::types::U16::new(value.try_into()?)),
+                    Default::default(),
+                )),
+                _ => bail!("Unsupported literal type: {literal_type} for array"),
+            }
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+
+    ensure!(len == hash.len(), "Expected to produce {len} elements, but produced {}", hash.len());
+
+    Ok(Plaintext::Array(hash, Default::default()))
+}
+
+fn to_circuit_array_u8<A: circuit::Aleo>(hash: &[circuit::Boolean<A>], len: usize) -> Result<circuit::Plaintext<A>> {
+    use circuit::{Eject, Inject};
+
+    ensure!(hash.len() % 8 == 0, "Expected hash length to be a multiple of 8, but found {}", hash.len());
+
+    let hash: Vec<_> = hash
+        .chunks(8)
+        .map(|element| {
+            let value = element
+                .iter()
+                .enumerate()
+                .fold(0, |acc, (i, bit)| if bit.eject_value() { acc | (1 << i) } else { acc });
+
+            let value = console::types::U8::new(value);
+            circuit::Plaintext::Literal(
```

Comment:
I don't believe this circuit is constrained sufficiently. You'll need to use the `from_bits` and `to_bits` utilities to do so.